### PR TITLE
Report ns

### PR DESCRIPTION
**Pull request by nick-pestell**
This PR adds the reporting of the number of missing sites 'Ns' in the pipeline genome to stats.

It also changes the names in stats.json from "masked_TPs" to "TPs in mask" etc.

### CODE REVIEW

File: sim/compare_snps.py [hunk 1]
```diff
@@ -18,25 +19,46 @@ def masked_positions(mask_filepath):
...
+        raise Exception("Cant Find the pipeline's consensus file!!")
     # Load
-    simulated = pd.read_csv(simulated_snps, delimiter='\t')
-    pipeline = pd.read_csv(pipeline_snps, delimiter='\t')
+    simulated_snps = pd.read_csv(simulated_snp_path,  delimiter='\t')
+    pipeline_snps = pd.read_csv(pipeline_snp_path, delimiter='\t')
+    pipeline_genome = load_consensus(pipeline_genome_path)
+
+    # Trim adapter sequences off pipeline genome
+    pipeline_genome = pipeline_genome[29:-31]
```
Comment (afishman) [inline sim/compare_snps.py:52]: Hardcoded values, where did you get these lengths from?

File: sim/compare_snps.py [hunk 2]
```diff
@@ -18,25 +19,46 @@ def masked_positions(mask_filepath):
...
-    simulated_pos = set(simulated['ref_start'].values)
-    pipeline_pos = set(pipeline['POS'].values)
+    simulated_pos = set(simulated_snps['ref_start'].values)
+    pipeline_pos = set(pipeline_snps['POS'].values)
+    
+    # Extract mask positions    
     masked_pos = set(masked_positions(mask_filepath))
 
+    # Extract N's positions in pipeline genome
+    n_pos = set([i+30 for i in range(len(pipeline_genome)) if pipeline_genome[i] == 'N'])
```
Comment (afishman) [inline sim/compare_snps.py:63]: another hardcoded value, based off the previous

File: sim/compare_snps.py [hunk 3]
```diff
@@ -18,25 +19,46 @@ def masked_positions(mask_filepath):
...
     return masked_pos
 
-def analyse(simulated_snps, pipeline_snps, mask_filepath):
+def analyse(results_path, sample_name, mask_filepath):
     """ Compare simulated SNPs data from simuG against btb-seq's snpTable.tab
         If adjust == True: applies the mask to simulated SNPs and pipeline SNPs
         Returns a dictionary of performance stats
     """
 
+    pipeline_directory = glob.glob(results_path + 'btb-seq-results/Results_simulated-reads_*')[0] + '/'
```
Comment (afishman) [inline sim/compare_snps.py:33]: Add a TODO to DRY this up with what's in `validator.py` we should probably load data into memory in validator.py, or pass the filepath in directly to this function (leaning towards the former)

File: sim/compare_snps.py (hunk 1 above)
Comment (nick-pestell) [reply sim/compare_snps.py:52]: These were from counting the Ns on the start and end of the consensus file. Where would I find the proper adapter sequence lengths?

File: sim/compare_snps.py (hunk 2 above)
Comment (nick-pestell) [reply sim/compare_snps.py:63]: This +30 is to acheive a consistent indexing with the positions in the map.txt and .tab files, which start at 30 (I guess due to the adapter sequence).

File: sim/compare_snps.py (hunk 1 above)
Comment (afishman) [reply sim/compare_snps.py:52]: ask Richard, you can put a TODO for now though if you like.

File: sim/compare_snps.py [hunk 4]
```diff
@@ -75,20 +103,19 @@ def analyse(simulated_snps, pipeline_snps, mask_filepath):
         "TP": tp,
         "FP": fp,
         "FN": fn,
-        "masked TPs": tp_in_mask,
-        "masked FPs": fp_in_mask, 
-        "masked FNs": fn_in_mask, 
+        "misssing sites": n,
+        "TPs in mask": tp_in_mask,
+        "FPs in mask": fp_in_mask, 
+        "FNs in mask": fn_in_mask,
```
Comment (afishman) [inline sim/compare_snps.py]: Please remove spaces in for the dictionary key names. Spaces are generally more awkward to deal with in this context

**Review by afishman — CHANGES_REQUESTED** [review]

File: sim/compare_snps.py (hunk 2 above)
Comment (afishman) [reply sim/compare_snps.py:63]: I see, perhaps just put a comment above to saying that, will make it easier to cleanup later

**Review by afishman — APPROVED** [review]
cool

**Pull request merged**